lockfile/src/csharp.rs

```diff
@@ -266,4 +268,24 @@ mod tests {
         let pkgs = CSProj.parse(include_str!("../../tests/fixtures/Calculator.csproj")).unwrap();
         assert!(!pkgs.is_empty());
     }
+
+    #[test]
+    fn test_overlapped_case() {
+        #[derive(Debug, Deserialize, PartialEq)]
+        struct Foo {
+            bar: Vec<usize>,
+            baz: String,
+        }
+        let s = r##"
+        <foo>
+            <bar>1</bar>
+            <bar>2</bar>
+            <baz>Hello, world</baz>
+            <bar>3</bar>
+            <bar>4</bar>
+        </foo>
+        "##;
+        let foo: Foo = quick_xml::de::from_str(s).unwrap();
+        assert_eq!(foo, Foo { bar: vec![1, 2, 3, 4], baz: "Hello, world".to_string() });
+    }
```

Comment:
Why are we testing `quick_xml` in our tests? This seems unrelated to our actual objectives.

Reply:
Ah, this is due to https://github.com/phylum-dev/cli/blob/9fa177b58862c68fa7db793182e0af5b87570a10/lockfile/src/csharp.rs#L147. We didn't have a test case that covered the use of [non_contiguous_seq_elements](https://docs.rs/serde-xml-rs/0.6.0/serde_xml_rs/de/struct.Deserializer.html#method.non_contiguous_seq_elements).  It's disabled by default but was specifically enabled so it was probably encountered at some point. `quick-xml` has [overlapped-lists](https://docs.rs/quick-xml/0.31.0/quick_xml/index.html#overlapped-lists) feature and I wanted to verify it worked accordingly.

Reply:
Is it possible to have a test more focused on our actual usage of this feature? I would have imagined it's part of the new test data you added.

Reply:
I haven't added any new test data for csharp other than this test. The other work is `SPDX` related, but I'll look at generating something closer to our usage or find something that has it.

Reply:
Updated the overlapped test case.